lib/ansible/modules/database/mysql/mysql_user.py

```diff
@@ -401,8 +401,15 @@ def user_mod(cursor, user, host, host_all, password, encrypted, new_priv, append
             # and in the new privileges, then we need to see if there's a difference.
             db_table_intersect = set(new_priv.keys()) & set(curr_priv.keys())
             for db_table in db_table_intersect:
-                priv_diff = set(new_priv[db_table]) ^ set(curr_priv[db_table])
-                if len(priv_diff) > 0:
+                if "ALL" in new_priv[db_table] and db_table == "*.*":
+                    version = get_db_version(cursor)
```

Comment:
I guess it would be best to check version + flavor (MySQL vs MariaDB)
Can't we enhance the version check function to include informations that would allow us to better handle this condition ?

Reply:
Mb I missed something, as I see, for now, we don't need to know which version is used except Mysql 8, so we don't need to parse is it MariaDB or not. But if you have a different point of view feel free to modify the code. By the way, I've made some fixes so now it should work with all MySQL versions and MariaDB.

Reply:
@azeliashchonak  Will the change affecting MySQL 8 be ported in MariaDB 10.3 or later ?
If so, we absolutely need to handle flavor + affected versions, and not just '8'.
If not, then fine with it as is.

Reply:
@bmalynovytch  Currently it's not ported. And we can't be sure it will or will not be. I guess we can now implement it as is. In case of further incompatibility to MariaDB this code can be easily adjusted. If you can propose better way of handling of current behavior - we're welcome.

Reply:
👍